src/librustc_typeck/check/mod.rs

```diff
@@ -1522,6 +1517,14 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
         self.tables.borrow_mut().node_types.insert(node_id, ty);
     }
 
+    #[inline]
+    pub fn write_ty_expr(&self, node_id: ast::NodeId, ty: Ty<'tcx>) {
```

Comment:
Do we need this method? Couldn't it just check if there's a `NodeExpr` in `tcx.map` for that ID?